src/models.py

```diff
@@ -1,6 +1,28 @@
+import re
 from enum import Enum
 
+from web3 import Web3
+
 
 class Network(Enum):
     MAINNET = 'mainnet'
     GCHAIN = 'gchain'
+
+
+class Account:
```

Comment:
I think for it to make sense to use the same class to store `token_address` as well as `receiver` it should probably be called `Address`.

Reply:
Or even EthereumAddress

Reply:
Since this script should also work on gnosis chain `EthereumAddress` seems too specific.

Reply:
Have refactored it as `Address`. However it will be a long while before we depend on any Gnosis Chain related dune data (since gas costs are essentially zero there).

Reply:
Actually I think it might be better to go with `HexAddress` some of the ethereum python libraries that have typing use Address for bytes and HexAddress for string. Note that I would use the types out of the box, but they don't seem to include validation.